test/IRGen/partial_apply.sil

```diff
@@ -437,3 +437,63 @@ bb0(%0 : $Int):
   %result = tuple ()
   return %result : $()
 }
+
+
+// Crash on partial apply of a generic enum.
+enum GenericEnum<T> {
+	case X(String)
```

Comment:
In Factor we solved this problem by making tabs an error in the lexer. I don't think that would sit well with swift-evolution though...